Model/Api/Tax.php

```diff
@@ -169,17 +181,53 @@ public function createInstorePickOption($totalsInformation, $currencyCode, $ship
         $storeAddress->setRegion($ship_to_store_option['address']['region']);
         $storeAddress->setPostalCode($ship_to_store_option['address']['postal_code']);
         
+        $ship_to_store_option = $this->getShippingDiscount($totalsInformation, $currencyCode, $ship_to_store_option);
+        
         $shipToStoreOption = $this->shipToStoreOptionFactory->create();
         $shipToStoreOption->setReference($ship_to_store_option['reference']);
-        $shipToStoreOption->setCost(CurrencyUtils::toMinor($totalsInformation->getShippingAmount(), $currencyCode));
+        $shipToStoreOption->setCost($ship_to_store_option['cost']);
         $shipToStoreOption->setStoreName($ship_to_store_option['store_name']);
         $shipToStoreOption->setAddress($storeAddress);
         $shipToStoreOption->setDistance($ship_to_store_option['distance']);
         $shipToStoreOption->setDistanceUnit($ship_to_store_option['distance_unit']);
-        $shipToStoreOption->setTaxAmount(CurrencyUtils::toMinor($totalsInformation->getShippingTaxAmount(), $currencyCode));
+        $shipToStoreOption->setTaxAmount($ship_to_store_option['tax_amount']);
 
         return $shipToStoreOption;
     }
+    
+    /**
+     * @param TotalsInterface $totalsInformation
+     * @param string $currencyCode
+     * @param array $shipping_option
+     * @return array
+     * @throws \Exception
+     */
+    protected function getShippingDiscount($totalsInformation, $currencyCode, $shipping_option)
+    {
+        $shippingDiscountAmount = $this->eventsForThirdPartyModules->runFilter("collectShippingDiscounts", $totalsInformation->getShippingDiscountAmount(), $this->quote, $this->quote->getShippingAddress());
+        if ($shippingDiscountAmount >= DiscountHelper::MIN_NONZERO_VALUE) {
+            $service = $shipping_option['service'] ?? '';
+            $shippingCost = $totalsInformation->getShippingAmount() - $shippingDiscountAmount;
+            $shippingRoundedCost = CurrencyUtils::toMinor($shippingCost, $currencyCode);
+            $diff = CurrencyUtils::toMinorWithoutRounding($shippingCost, $currencyCode) - $shippingRoundedCost;
+            $taxAmount = round(CurrencyUtils::toMinorWithoutRounding($totalsInformation->getShippingTaxAmount(), $currencyCode) + $diff);
+            if ($shippingRoundedCost == 0) {
+                $service .= ' [free&nbsp;shipping&nbsp;discount]';
+            } else {
+                $discount = $this->priceHelper->currency($shippingDiscountAmount, true, false);
+                $service .= " [$discount" . "&nbsp;discount]";
+            }
+            $shipping_option['service'] = html_entity_decode($service);
```

Comment:
Since we are adding a suffix either way, let's decode (only) the service before.

Reply:
@danac-gs I believe this is to convert the '&nbsp;' entity to ASCII code 160 properly. Actually this was created by you https://github.com/BoltApp/bolt-magento2/commit/be3986e883835acd2cf244c3220fa9343862fabf#diff-11cba1d080bcf8a37d870a6bffb136b052d3be87e055ac78ff8ff87fa20f5355R568